### PR TITLE
Pyssht

### PR DESCRIPTION
**Pull request by steven-murray**
Remove dependence on pyssht. This should not require the merging of the ssht_numba PR -- it just changes a couple of calls to use ssht_numba instead.

### CODE REVIEW

File: src/spin1_beam_model/cst_processing.py [hunk 1]
```diff
@@ -132,12 +132,11 @@ def compute_spin1_harmonics(self, filter_cut_factor=1.0):
             pos1_E = (Et + 1j * Ep) / np.sqrt(2.0)
             neg1_E = (Et - 1j * Ep) / np.sqrt(2.0)
 
-            pos1_Elm = pyssht.forward(
-                pos1_E, self.L_data, Spin=1, Method="MWSS", Reality=False
-            )
-            neg1_Elm = pyssht.forward(
-                neg1_E, self.L_data, Spin=-1, Method="MWSS", Reality=False
-            )
+            pos1_Elm = np.empty([self.L_data * self.L_data], dtype=complex)
```
Comment (plaplant) [inline src/spin1_beam_model/cst_processing.py:135]: Can we give (brief) insight into these magic numbers with a comment? If that'd be too long or involved, we can also just point to the documentation of `mw_forward_sov_conv_sym_ss` (assuming docs exist...)

File: src/spin1_beam_model/jones_matrix_field.py [hunk 2]
```diff
@@ -338,12 +336,11 @@ def splines_from_harmonics(pos1_Elm, neg1_Elm, L_synth):
             pos1_Elm_pad = pad_Elm(pos1_Elm, L_synth)
             neg1_Elm_pad = pad_Elm(neg1_Elm, L_synth)
 
-            pos1_E = pyssht.inverse(
-                pos1_Elm_pad, L_synth, Spin=1, Method="MWSS", Reality=False
-            )
-            neg1_E = pyssht.inverse(
-                neg1_Elm_pad, L_synth, Spin=-1, Method="MWSS", Reality=False
-            )
+            pos1_E = np.empty([L_synth + 1, 2 * L_synth], dtype=np.complex)
```
Comment (plaplant) [inline src/spin1_beam_model/jones_matrix_field.py:339]: Same comment as above.

**Review by plaplant — COMMENTED** [review]
Thanks @steven-murray! These changes seem straightforward enough. I just had a couple of questions around documentation.

File: src/spin1_beam_model/cst_processing.py (hunk 1 above)
Comment (steven-murray) [reply src/spin1_beam_model/cst_processing.py:135]: Docs most definitely do not exist (yet)! I don't really know what the numbers mean, I just took them from the code of ssht itself. Not really sure where to point, or what to say! The relevant line is https://github.com/UPennEoR/ssht/blob/public/src/python/pyssht.pyx#L258

File: src/spin1_beam_model/jones_matrix_field.py (hunk 2 above)
Comment (steven-murray) [reply src/spin1_beam_model/jones_matrix_field.py:339]: ditto

File: src/spin1_beam_model/cst_processing.py (hunk 1 above)
Comment (plaplant) [reply src/spin1_beam_model/cst_processing.py:135]: Okay, no worries then. Thanks for tracking it down. We can just add it to our ongoing crusade to improve the documentation on the repo.

**Review by plaplant — APPROVED** [review]

**Pull request merged**